pinot-controller/src/test/java/com/linkedin/pinot/controller/api/resources/PinotInstanceRestletResourceTest.java

```diff
@@ -63,6 +63,9 @@ public void testInstanceListingAndCreation() throws Exception {
     serverInstance.put("host", "2.3.4.5");
     sendPostRequest(_controllerRequestURLBuilder.forInstanceCreate(), serverInstance.toString());
 
+    // Sleep 0.5s for cache accessor to update cache.
+    Thread.sleep(500L);
```

Comment:
Is the sleep required only here?

Also, a better pattern might be to use TestUtils.waitForCondition instead of a simple sleep if we are to wait on results of asynchronous operations to get reflected.

Reply:
See example here: https://github.com/apache/incubator-pinot/blob/master/pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/OfflineClusterIntegrationTest.java#L162

Reply:
Done. Thanks for the tip!